packages/multichain-account-service/src/providers/EvmAccountProvider.test.ts

```diff
@@ -223,17 +273,74 @@ describe('EvmAccountProvider', () => {
     ).rejects.toThrow('Internal account does not exist');
   });
 
-  it('discover accounts', async () => {
+  it('discover accounts at the next group index', async () => {
     const { provider } = setup({
-      accounts: [], // No accounts by defaults, so we can discover them
+      accounts: [],
     });
 
-    // TODO: Update this once we really implement the account discovery.
+    const expectedAccount = {
+      id: expect.any(String),
+      address: `${MOCK_HD_ACCOUNT_1.address}0`,
+      options: {
+        entropy: {
+          id: MOCK_HD_KEYRING_1.metadata.id,
+          type: 'mnemonic',
+          groupIndex: 0,
+          derivationPath: '',
+        },
+      },
+      methods: [...ETH_EOA_METHODS],
+      scopes: [EthScope.Eoa],
+      type: EthAccountType.Eoa,
+      metadata: {
+        name: 'Account 1',
+        keyring: { type: KeyringTypes.hd },
+        importTime: 0,
+        lastSelected: 0,
+        nameLastUpdatedAt: 0,
+      },
+    };
```

Comment:
Nit: We could have used the `MOCK_HD_ACCOUNT_1` for this maybe?

Like:
```ts
const expectedAccount = {
  ...MockAccountBuilder.from(MOCK_HD_ACCOUNT_1).withAddressSuffix(0).get(),
  id: expect.any(String),
}
```

Reply:
True